src/plugins/dummy-plugin/DummyPlugin.ts

```diff
@@ -4,46 +4,64 @@ import { IUserDao } from "../../daos/IUserDao";
 import { ISessionDao } from "../../daos/ISessionDao";
 import { IRequestDao } from "../../daos/IRequestDao";
 import { Request } from "../../model/Request";
+import {DaoFactory} from "../../daos/DaoFactory"
+import { MariaDBDaoFactory } from "../../daos/maria/MariaDBDaoFactory";
+import { MongoDBDaoFactory } from "../../daos/mongo/MongoDBDaoFactory";
 
 export class DummyPlugin implements IPersistenceProviderPlugin {
-	gameDao: IGameDao = {
-		saveGame() {},
-		getGame() {
-			return JSON;
-		},
-	};
-
-	userDao: IUserDao = {
-		saveUser() {},
-		getUser(id: number) {
-			return {username: 'Betty the Bot', password: 'Super Secr3t'};
-		},
-		getAllUsers() {
-			return []
+	factory:DaoFactory;
+	gameDao:IGameDao;
+	userDao:IUserDao;
+	requestDao:IRequestDao;
+	sessionDao:ISessionDao;
+	constructor(pluginType:boolean){
+		
+		if(pluginType){
+			this.factory=new MariaDBDaoFactory();
+		} else{
+			this.factory=new MongoDBDaoFactory();
 		}
-	};
+		this.gameDao=this.factory.createGameDao();
+		this.userDao=this.factory.createUserDao();
+		this.sessionDao=this.factory.createSessionDao();
+		this.requestDao=this.factory.createRequestDao();
+	}
+	// gameDao: IGameDao = {
```

Comment:
Are we going to be using this commented out stuff later?

Reply:
Yeah I'll take that out.